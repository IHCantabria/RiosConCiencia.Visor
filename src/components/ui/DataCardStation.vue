<template>
  <div class="overviewcard">
    <div class="overviewcard__description">
      <vs-list class="additional-Info">
        <vs-list-item class="additional-Info__Icon">
          <i
            :class="[openLicense ? 'fas fa-unlock' : 'fas fa-lock']"
            :title="licenseType"
          ></i>
        </vs-list-item>
        <vs-list-item title="Latitude">
          <span>{{ latitude }}</span></vs-list-item
        >
        <vs-list-item title="Longitude"
          ><span>{{ longitude }}</span></vs-list-item
        >
        <vs-list-item title="Height">
          <span>{{ height }}</span></vs-list-item
        >
        <vs-list-item title="Proyect Name"
          ><span>{{ proyectName }}</span></vs-list-item
        >
        <vs-list-item title="Location"
          ><span>{{ location }}</span></vs-list-item
        >
      </vs-list>
      <vs-list class="variables-Info">
        <vs-list-header
          icon="info"
          title="Variables Info"
          color="primary"
        ></vs-list-header>
        <vs-list-item
          v-for="variable in variables"
          :key="variable.code"
          :title="variable.name"
        >
          <span
            ><strong>observations</strong>: {{ variable.observations }}</span
          >
          <span><strong>Units</strong>: {{ variable.units }}</span>
        </vs-list-item>
      </vs-list>
    </div>
  </div>
</template>
<script>
export default {
  props: {
    openLicense: {
      type: Boolean,
      required: false,
      default: () => {
        return false;
      }
    },
    licenseType: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    longitude: {
      type: Number,
      required: false,
      default: () => {
        return 0;
      }
    },
    latitude: {
      type: Number,
      required: false,
      default: () => {
        return 0;
      }
    },
    height: {
      type: Number,
      required: false,
      default: () => {
        return 0;
      }
    },
    proyectName: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    location: {
      type: String,
      required: false,
      default: () => {
        return "";
      }
    },
    variables: {
      type: Array,
      required: false,
      default: () => {
        return [];
      }
    }
  }
};
</script>
<style lang="scss" scoped>
.overviewcard {
  border-top: solid $color-primary;
  border-bottom: solid $color-primary;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  background-color: $color-very-light-grey;

  &__description {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
  }
}
</style>
