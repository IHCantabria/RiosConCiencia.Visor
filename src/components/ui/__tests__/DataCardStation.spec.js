import { shallowMount, createLocalVue } from "@vue/test-utils";
import DataCardStation from "../DataCardStation.vue";
import Vuesax from "vuesax";
import merge from "lodash.merge";

const localVue = createLocalVue();
localVue.use(Vuesax);

function createWrapper(overrides) {
  const defaultMountingOptions = { localVue };
  return shallowMount(
    DataCardStation,
    merge(defaultMountingOptions, overrides)
  );
}

describe("dataCardStation.vue", () => {
  //SNAPSHOT TESTS
  it("renders correctly default values", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly with full info", () => {
    expect.assertions(1);
    const wrapper = createWrapper({
      propsData: {
        openLicense: true,
        licenseType: "open",
        longitude: 10,
        latitude: 30,
        height: 40,
        proyectName: "proyect test 1",
        location: "location test 1",
        variables: [
          {
            code: 1,
            name: "name test 1",
            description: "description test 1",
            units: "m"
          },
          {
            code: 2,
            name: "name test 2",
            description: "description test 2",
            units: "s"
          }
        ]
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
