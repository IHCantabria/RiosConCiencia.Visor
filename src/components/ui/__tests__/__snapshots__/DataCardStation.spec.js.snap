// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataCardStation.vue renders correctly default values 1`] = `
<div
  class="overviewcard"
>
  <div
    class="overviewcard__description"
  >
    <vs-list-stub
      class="additional-Info"
    >
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <i
          class="fas fa-lock"
          title=""
        />
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Latitude"
      >
        <span>
          0
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Longitude"
      >
        <span>
          0
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Height"
      >
        <span>
          0
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Proyect Name"
      >
        <span>
          
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Location"
      >
        <span>
          
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
     
    <vs-list-stub
      class="variables-Info"
    >
      <vs-list-header-stub
        color="primary"
        icon="info"
        iconpack="material-icons"
        title="Variables Info"
      />
       
    </vs-list-stub>
  </div>
</div>
`;

exports[`dataCardStation.vue renders correctly with full info 1`] = `
<div
  class="overviewcard"
>
  <div
    class="overviewcard__description"
  >
    <vs-list-stub
      class="additional-Info"
    >
      <vs-list-item-stub
        class="additional-Info__Icon"
        iconpack="material-icons"
      >
        <i
          class="fas fa-unlock"
          title="open"
        />
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Latitude"
      >
        <span>
          30
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Longitude"
      >
        <span>
          10
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Height"
      >
        <span>
          40
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Proyect Name"
      >
        <span>
          proyect test 1
        </span>
      </vs-list-item-stub>
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="Location"
      >
        <span>
          location test 1
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
     
    <vs-list-stub
      class="variables-Info"
    >
      <vs-list-header-stub
        color="primary"
        icon="info"
        iconpack="material-icons"
        title="Variables Info"
      />
       
      <vs-list-item-stub
        iconpack="material-icons"
        title="name test 1"
      >
        <span>
          <strong>
            observations
          </strong>
          : 
        </span>
         
        <span>
          <strong>
            Units
          </strong>
          : m
        </span>
      </vs-list-item-stub>
      <vs-list-item-stub
        iconpack="material-icons"
        title="name test 2"
      >
        <span>
          <strong>
            observations
          </strong>
          : 
        </span>
         
        <span>
          <strong>
            Units
          </strong>
          : s
        </span>
      </vs-list-item-stub>
    </vs-list-stub>
  </div>
</div>
`;
